src/coniql/strawberry_schema.py

```diff
@@ -0,0 +1,263 @@
+
+
+async def resolver_value(root: DeferredChannel) -> Optional[TypeChannelValue]:
+    channel = await root.get_channel()
+    return channel.get_value()
+
+
+async def resolver_time(root: DeferredChannel) -> Optional[TypeChannelTime]:
+    channel = await root.get_channel()
+    return channel.get_time()
+
+
+async def resolver_status(root: DeferredChannel) -> Optional[TypeChannelStatus]:
+    channel = await root.get_channel()
+    return channel.get_status()
+
+
+async def resolver_display(root: DeferredChannel) -> Optional[ChannelDisplay]:
+    channel = await root.get_channel()
+    return channel.get_display()
+
+
+@strawberry.type
+class Channel:
+    """
+    A single value with associated time, status and metadata. These values
+    can be Null so that in a subscription they are only updated on change
+    """
+
+    # ID that uniquely defines this Channel, normally a PV
+    id: strawberry.ID
+    # The current value of this channel
+    value: Optional[ChannelValue] = strawberry.field(resolver=resolver_value)
+    # When was the value last updated
+    time: Optional[ChannelTime] = strawberry.field(resolver=resolver_time)
+    # Status of the connection, whether is is mutable, and alarm info
+    status: Optional[ChannelStatus] = strawberry.field(resolver=resolver_status)
+    # How should the Channel be displayed
+    display: Optional[ChannelDisplay] = strawberry.field(resolver=resolver_display)
+
+
+def get_channel(id: strawberry.ID, timeout: float = 5.0) -> TypeChannel:
+    return GetChannel(id, timeout, store_global)
+
+
+@strawberry.type
+class Query:
+    # Get the current value of a Channel
+    getChannel: Channel = strawberry.field(resolver=get_channel)
+
+
+async def subscribe_channel(id: strawberry.ID) -> AsyncGenerator[TypeChannel, None]:
+    """Subscribe to changes in top level fields of Channel,
+    if they haven't changed they will be Null"""
+    store: PluginStore = store_global
+    plugin, channel_id = store.plugin_config_id(id)
+    # Remove the transport prefix from the read pv
+    pv = store.transport_pv(id)[1]
+    async for channel in plugin.subscribe_channel(pv):
+        yield SubscribeChannel(channel_id, channel)
+
+
+@strawberry.type
+class Subscription:
+    """Tell mypy to ignore this line as it complains 'expression
+    has type "T", variable has type "Channel"'. Possibly a Strawberry issue
+    as strawberry.subscription is a partial function"""
+
+    subscribeChannel: Optional[Channel] = strawberry.subscription(
+        resolver=subscribe_channel
+    )  # type: ignore
+
+
+@strawberry.type
+class Mutation:
+    @strawberry.mutation
+    async def putChannels(
+        self,
+        ids: List[strawberry.ID],
+        values: List[str],
+        timeout: float = 5.0,
+    ) -> Sequence[Channel]:
+        """Put a list of values to a list of Channels"""
+        store: PluginStore = store_global
+        pvs = []
+        plugins = set()
```

Comment:
Can these both have type hints added? Especially the set.

Reply:
I've added type hints for these.